datalad/local/add_readme.py

```diff
@@ -82,6 +89,20 @@ def __call__(dataset, filename='README.md', existing='skip'):
         # unlock, file could be annexed
         if lexists(filename):
             dataset.unlock(filename)
+        if not lexists(filename) and not annex:
```

Comment:
Actually, you're not doing what the doc reads: `--annex` doesn't make sure it's annexed (while not giving it *does* make sure it's going into git). Instead, with `--annex` what happens is, whatever the existing configuration of the datasets implies. That's not exactly the same thing.

Reply:
That is true! Do you have an idea for a better parameter name?

Reply:
``--honor-config``?

Reply:
Well, honoring the config should be expected default anyway. Would be strange to have that as an option. Implementation just needs an `else` where we record the opposite in `.gitattributes` explicitly.

Reply:
Or: Don't make it an `else` really, but have the block `if not lexists(filename):` and make the value for `annex.alrgefiles` conditional:

` ... {'annex.largefiles': 'anything' if annex else 'nothing'} ...`

Reply:
ok, I have implemented this. Weirdest .gitattribute configuration I have written so far :D